### PR TITLE
Introduce CONTRIBUTING.md and GitHub issue templates

### PR DESCRIPTION
**Pull request by rybak**
### TODO

- [x] add issue template for bug reports

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+- If anything isn't clear in the code, feel free to create an
+  [issue on GitHub][GitHubIssues] or [on GitLab][GitLabIssues] with a question.
+
+- If you want to add support for a hosting provider which is not yet supported:
+
+  1. Before making code changes, please create an [issue on GitHub][GitHubIssues]
+     or [on GitLab][GitLabIssues] to track the ongoing work.
```
Comment (dtruebin) [inline CONTRIBUTING.md:7]: Is it possible that "closes #123" in a message of a commit pushed to master will close the issue with that number in both GitHub and GitLab?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+     `GitHosting` can be overridden.
+
+- Please keep the project's [goals and non-goals](./README.md#goals) in mind.
+
+- If your pull request requires changes in class `GitHosting` or top-level
+  functions, please be aware that ensuring that the userscript still works on
+  supported Git hostings is the top priority.  This may take a while to do due
+  to limited access to some of the supported hostings.
+
+- Before merging, your commits might be reorganized.
```
Comment (dtruebin) [inline CONTRIBUTING.md:19]: This may be not clear enough, but I suppose you can cross that bridge when you get there :)

**Review by dtruebin — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (rybak) [reply CONTRIBUTING.md:7]: Yes, but I won't be using just the number `#123`. I'm already using full URLs, GitHub understands them: [example](https://github.com/rybak/copy-commit-reference-userscript/commit/c938c24853076ca4b3ed602a709cd9914ca7efff.patch).

**Comment by rybak** [comment]
Force pushed to squash fixup commits. Range diff:

```
1:  5f67321 ! 1:  50c999a README: describe goals and non-goals
    @@ README.md: License, Version 3.  See [LICENSE.txt](LICENSE.txt) for details.
     +- Provide a way of producing human-readable HTML links, which are paste-able in
     +  rich text editors of wikis and IM apps, like Confluence and Slack.
     +  - Where possible, support HTML links to issue trackers and pull requests.
    -+- Use native-like UI elements (buttons, links, ands tooltips) which don't look
    ++- Use native-like UI elements (buttons, links, and tooltips) which don't look
     +  out of place on the websites of supported Git hostings.
     +- Provide an abstract class `GitHosting` which:
     +  - is easy to subclass
    @@ README.md: License, Version 3.  See [LICENSE.txt](LICENSE.txt) for details.
     +
     +- Adding buttons to copy full commit hashes or other useful strings from the web
     +  UIs of Git hostings.
    -+- Overwritting or replacing existing UI.  For example, native buttons to copy
    ++- Overwriting or replacing existing UI.  For example, native buttons to copy
     +  hashes of commits should continue to work.
     +
      [GreasyForkInstallsBadge]: https://img.shields.io/badge/dynamic/json?style=flat&color=670000&label=Greasy%20Fork&query=total_installs&suffix=%20installs&url=https%3A%2F%2Fgreasyfork.org%2Fscripts%2F473195.json
2:  caad623 ! 2:  673ab5b CONTRIBUTING: new documentation file
    @@ CONTRIBUTING.md (new)
     +  supported Git hostings is the top priority.  This may take a while to do due
     +  to limited access to some of the supported hostings.
     +
    -+- Before merging, your commits might be reorganized.
    -+
     +[GitHubIssues]: https://github.com/rybak/copy-commit-reference-userscript/issues
     +[GitLabIssues]: https://gitlab.com/andrybak/copy-commit-reference-userscript/-/issues
     
3:  e306851 = 3:  a45ed78 .github: add issue template "Support new hosting"
4:  06f41cb < -:  ------- fixup! README: describe goals and non-goals
5:  2e97594 = 4:  e391cce .github: add issue template "Bug report"
```

**Pull request merged**